src/main/java/com/example/shoppingonlybackend/controller/ProductController.java

```diff
@@ -68,4 +68,25 @@ public ResponseEntity<?> delete(@PathVariable int id) {
         }
         return ResponseEntity.notFound().build();
     }
+    
+    @PreAuthorize("hasRole('MEMBER') or hasRole('ADMIN')")
+    @GetMapping("/searchByName")
+    public ResponseEntity<?> searchByName(@RequestParam String keyword) {
+        List<Product> products = productService.searchProductsByName(keyword);
+        return ResponseEntity.ok(products);
```

Comment:
```suggestion
        if (products.isEmpty())
                return ResponseEntity.noContent();

        return ResponseEntity.ok(products);
```
Veri yoksa 200 dönmesin.